happytransformer/happy_word_prediction.py

```diff
@@ -55,11 +57,16 @@ def predict_mask(self, text: str, targets: Optional[List[str]] = None, top_k: in
             for answer in answers
         ]
 
-    def train(self, input_filepath, args):
-        raise NotImplementedError("train() is currently not available")
+    def train(self, input_filepath, args=ARGS_WP_TRAIN):
+        self._trainer.train(input_filepath=input_filepath, args=args)
 
-    def eval(self, input_filepath):
-        raise NotImplementedError("eval() is currently not available")
+    def eval(self, input_filepath, args) -> EvalResult:
+        dataclass_args = self._create_args_dataclass(default_dic_args=ARGS_WP_EVAl,
```

Comment:
I'd argue that `dataclass_args=WpEvalArgs(**ARGS_WP_EVAL,**args)` is significantly more readable.

Also why not just bake the entries of `ARGS_WP_EVAL` as the defaults of `WpEvalArgs`?

From https://docs.python.org/3/library/dataclasses.html

```
@dataclass
class C:
    a: int       # 'a' has no default value
    b: int = 0   # assign a default value for 'b'
```

Reply:
Regarding `dataclass_args=WpEvalArgs(**ARGS_WP_EVAL,**args)`, this results in a TypeError when ARGS_WP_EVAL and args contain the same key. 

As for your comment about adding default values for WpEvalArgs, I wanted to avoid duplicating default values. With the current implementation, the default values are only stored in  ARGS_WP_EVALl which I think improves maintainability as we don't need to keep track of the default values in two locations.

Reply:
```
@dataclass
class WPEvalArgs:
    preprocessing_processes: int = 1
    mlm_probability: float = 0.15
    line_by_line: bool = False
```

Better solution.

Then init via `WpEvalArgs(**user_args)`

Reply:
By using `WpEvalArgs(**user_args)`, it results in the code under the HappyTransformer class being cleaner. However, as discussed, it would result in us keeping track of default values in two separate places for each eval task. 

So I think the current implementation is better because of that.

Reply:
"keeping track of default values in two separate places for each eval task" => elaborate on this. What requires you to store the values elsewhere?

Reply:
We'll show the users the ARGS_WP_EVAl dictionary in the documentation as an example of all of the potential/default settings.

The users can also import this dictionary, make changes as they please, and then provide it to HappyWordPrediction.eval()

Reply:
1) Users can easily see the settings by `print(WpEvalArgs())`
2) If we really need a dict, then this can be used https://docs.python.org/3/library/dataclasses.html#dataclasses.asdict

Reply:
Okay, I agree using a dataclass is more professional than using a dictionary. 

How about we allow the users to use either or, but recommend the dataclass version in the documentation? So, the HappyWordPrediction.eval() args value can be of type dictionary or of QAEvalArgs. 

I think we should avoid breaking changes, so we should keep on  supporting dictionaries. Maybe we can add a warning if they use a dictionary that when version 3.0.0 we'll stop supporting them.